File: verification/context-compression/fixtures/semantic-rich.java
import java.util.List;

public class Worker extends BaseWorker implements Runner {
    private final Repository repository;

    public Worker(Repository repository) {
        this.repository = repository;
    }

    public String run(String value) {
        audit(value);
        return repository.find(value);
    }
}

interface Runner extends BaseRunner {
    String run(String value);
}

File: verification/context-compression/fixtures/semantic-rich.cs
using System;

public interface IRunner : IBaseRunner
{
    string Run(string value);
}

public class Worker : BaseWorker, IRunner
{
    private readonly IRepository repository;

    public Worker(IRepository repository)
    {
        this.repository = repository;
    }

    public string Run(string value)
    {
        Console.WriteLine(value);
        return repository.Find(value);
    }
}
